development/workflow.rst

```diff
@@ -0,0 +1,668 @@
+
+Documentation edits and additions are the only scenarios where working directly on ``master`` and by-passing the code review process is permitted.
+In most cases, documentation writing benefits from peer editing (code review) and *can* be done on a ticket branch.
+
+The Git history of ``master`` **must never be re-written** with force pushes.
+
+.. _git-branch-user:
+
+User branches
+-------------
+
+You can do experimental, proof-of-concept work in 'user branches.'
+
+These branches are named
+
+.. code-block:: text
+
+   u/{{username}}/{{topic}}
+
+User branches can be pushed to GitHub to enable collaboration and communication.
+Before offering unsolicited code review on your colleagues' user branches, remember that the work is intended to be an early prototype.
+
+Developers can feel free to rebase and force push work to their personal user branches.
+
+A user branch *cannot* be merged into master; it must be converted into a *ticket branch* first.
+
+.. _git-branch-ticket:
+
+Ticket branches
+---------------
+
+Ticket branches are associated with a JIRA ticket.
+Only ticket branches can be merged into ``master``.
+(In other words, developing on a ticket branch is the only way to record earned value for code development.)
+
+If the JIRA ticket is named ``DM-NNNN``, then the ticket branch will be named
+
+.. code-block:: text
+
+   tickets/DM-NNNN
+
+A ticket branch can be made by branching off an existing user branch.
+This is a great way to formalize and shape experimental work into an LSST software contribution.
+
+When code on a ticket branch is ready for review and merging, follow the :ref:`code review process documentation <workflow-code-review>`.
+
+.. _git-branch-sims:
+
+Simulations branches
+--------------------
+
+The LSST Simulations team uses a different branch naming scheme:
+
+.. code-block:: text
+
+   feature/SIM-NNN-{{feature-summary}}
+
+.. _review-preparation:
+
+Review Preparation
+==================
+
+When development on your ticket branch is complete, we use a standard process for reviewing and merging your work.
+This section describes how to prepare your work for review.
+
+.. _workflow-pushing:
+
+Pushing code
+------------
+
+We recommend that you organize commits, improve commit messages, and ensure that your work is made against the latest commits on ``master`` with an `interactive rebase <https://help.github.com/articles/about-git-rebase/>`_.
+A common pattern is:
+
+.. code-block:: bash
+
+   git checkout master
+   git pull
+   git checkout tickets/DM-NNNN
+   git rebase -i master
+   # interactive rebase
+   git push --force
+
+.. _workflow-testing:
+
+Testing with Jenkins
+--------------------
+
+Use `Jenkins at ci.lsst.codes <https://ci.lsst.codes/job/stack-os-matrix/build?delay=0sec>`_ to run the Stack's tests with your ticket branch work.
+Jenkins finds, builds, and tests your work according to the name of your ticket branch; Stack repositories lacking your ticket branch will fall back to ``master``.
+
```

Comment:
I think a screenshot would be helpful here.  Also a note about github auth against lsst orgs being required in order to run a build.